code/go/0chain.net/smartcontract/dbs/event/search.go

```diff
@@ -0,0 +1,40 @@
+package event
+
+import (
+	"database/sql"
+	"strconv"
+)
+
+// GetGenericSearchType returns kind of search query
+func (edb *EventDb) GetGenericSearchType(query string) (string, error) {
+	var queryType string
+	var round int
+
+	round, err := strconv.Atoi(query)
+	if err != nil {
+		round = -1
+	}
+
+	res := edb.Store.Get().Raw(
+		"SELECT "+
+			"CASE "+
+			"WHEN EXISTS (SELECT Id FROM TRANSACTIONS WHERE HASH = @query) "+
+			"THEN 'TransactionHash' "+
+			"WHEN EXISTS (SELECT Id FROM BLOCKS WHERE HASH = @query) "+
+			"THEN 'BlockHash' "+
+			"WHEN EXISTS (SELECT Id FROM USERS WHERE USER_ID = @query) "+
+			"THEN 'UserId' "+
+			"WHEN EXISTS (SELECT Id FROM BLOCKS WHERE ROUND = @round) "+
+			"THEN 'BlockRound' "+
+			"WHEN EXISTS (SELECT Id FROM WRITE_MARKERS WHERE CONTENT_HASH = @query) "+
+			"THEN 'ContentHash' "+
+			"WHEN EXISTS (SELECT Id FROM WRITE_MARKERS WHERE NAME = @query) "+
+			"THEN 'FileName' "+
+			"ELSE 'Not Found' "+
+			"END AS queryType",
```

Comment:
hmm, I'm not a fan of put generic search in sharders, it should be a higher level function that should not be provided directly by the sharders,  otherwise it may cause performance issue for sharders, such as if I provide none exist query string every time, then the sharder will need to check all the tables to see if the query data exists. 

Instead, for the generic search function that the web app needed, we might need to parse the query string first to get the query type and data, and then call specific API to get the data respectively. 

The `query` format could be like the github filters that we can see when need to search pull requests or issues, such as `is:open is:pr author:peterlimg`, and for us, the query string could be for example: `block:af204cfd0e3a7dea37d168acbccad252c59d335dabf5647cb55d35f29e5caba2`, `round:1000`, `txn:af204cfd0e3a7dea37d168acbccad252c59d335dabf5647cb55d35f29e5caba2`, etc.

Reply:
> mm, I'm not a fan of putting generic search in sharders, it should be a higher level function that should not be provided directly by the sharders, otherwise it may cause performance issue for sharders, such as if I provide none exist query string every time, then the sharder will need to check all the tables to see if the query data exists.

not sure it is huge performance loss, it is 3 lookups to indexes, should be fast, much faster than aggregate functions for example. @peterlimg I agree with you that searching is not sharders responsibility and we might need some other indexing solution, such as elastic or similar, but this exact query will work fast